pyproject.toml

```diff
@@ -65,10 +65,32 @@ fallback_version = "unknown"
 # docstring-code-format = true
 # docstring-code-line-length = "dynamic"
 
+
 [tool.ruff.lint]
 select = [
+  # "ARG",  # flake8-unused-arguments IMPORTANT
+  # "B",    # flake8-bugbear IMPORTANT
```

Comment:
Why are these marked important?

Reply:
Bugbear because I feel it's really effective at catching latent bugs and I _think_ it detected some stuff that concerned me (I don't remember the specifics so I won't commit to this :laughing:). unused-args detected several cases of functions with unused arguments, and while that's not the end of the world, I personally find unused arguments act as red herrings when reading code, which can cause us to write bugs in the future!

Reply:
Is this list the full list of linting options ruff has, and the ones that are not commented are the ones it will address (i.e. change)?

I like the idea of eventually including the options you've noted as IMPORTANT as ways to improve code as we go (or they could be part of a targeted effort). Perhaps labeling them as "DEV-GOAL" or "PLANNED-FOR-FUTURE" is a clearer way to indicate this in the meantime?

Reply:
I believe it was the full list at one time, but I copied this from another project. It's likely there are new rulesets that aren't included. The ones that are not commented are the ones that will be checked and changed, correct.

Good call on clearer wording! Would you prefer to have this reflected in the comment, or as a GitHub issue to enable those rules? I have no strong preference, I think I did it the way I did because I "flowing" at the time and wrote it down in situ to think about more later :)

Reply:
> I believe it was the full list at one time, but I copied this from another project.

Does it then make sense to just remove the ones that are commented that we're not "planning" for future dev?

> Would you prefer to have this reflected in the comment, or as a GitHub issue to enable those rules?

Personally I'm happy with the comment (I've loosely used "DevGoal" like this throughout the code, so I can do a search and find things I hoped/planned to do in the past). For a while I also added things as issues, but because I was the primary contributor it meant I was spending a lot of time on admin of the issues instead of more dev work, so I started being more selective about when I opened an issue. Not the most helpful answer, but if you have a personal preference feel free to exercise that!

Reply:
>Does it then make sense to just remove the ones that are commented that we're not "planning" for future dev?

Personally, on most projects I have all of these rules on, and I think it's not unrealistic to target that for v2!

Reply:
a4cf9df

Happy to make more changes, let me know your thoughts!